avm/res/event-hub/namespace/eventhub/main.bicep

```diff
@@ -0,0 +1,265 @@
+
+@description('Optional. Name for capture destination.')
+param captureDescriptionDestinationName string = 'EventHubArchive.AzureBlockBlob'
+
+@description('Optional. Blob naming convention for archive, e.g. {Namespace}/{EventHub}/{PartitionId}/{Year}/{Month}/{Day}/{Hour}/{Minute}/{Second}. Here all the parameters (Namespace,EventHub .. etc) are mandatory irrespective of order.')
+param captureDescriptionDestinationArchiveNameFormat string = '{Namespace}/{EventHub}/{PartitionId}/{Year}/{Month}/{Day}/{Hour}/{Minute}/{Second}'
+
+@description('Optional. Blob container Name.')
+param captureDescriptionDestinationBlobContainer string = ''
+
+@description('Optional. Resource ID of the storage account to be used to create the blobs.')
+param captureDescriptionDestinationStorageAccountResourceId string = ''
+
+@description('Optional. A value that indicates whether capture description is enabled.')
+param captureDescriptionEnabled bool = false
+
+@description('Optional. Enumerates the possible values for the encoding format of capture description. Note: "AvroDeflate" will be deprecated in New API Version.')
+@allowed([
+  'Avro'
+  'AvroDeflate'
+])
+param captureDescriptionEncoding string = 'Avro'
+
+@description('Optional. The time window allows you to set the frequency with which the capture to Azure Blobs will happen.')
+@minValue(60)
+@maxValue(900)
+param captureDescriptionIntervalInSeconds int = 300
+
+@description('Optional. The size window defines the amount of data built up in your Event Hub before an capture operation.')
+@minValue(10485760)
+@maxValue(524288000)
+param captureDescriptionSizeLimitInBytes int = 314572800
+
+@description('Optional. A value that indicates whether to Skip Empty Archives.')
+param captureDescriptionSkipEmptyArchives bool = false
+
+@allowed([
+  'Compact'
+  'Delete'
+])
+@description('Optional. Retention cleanup policy. Enumerates the possible values for cleanup policy.')
+param retentionDescriptionCleanupPolicy string = 'Delete'
+
+@minValue(1)
+@maxValue(168)
+@description('Optional. Retention time in hours. Number of hours to retain the events for this Event Hub. This value is only used when cleanupPolicy is Delete. If cleanupPolicy is Compact the returned value of this property is Long.MaxValue.')
+param retentionDescriptionRetentionTimeInHours int = 1
+
+@minValue(1)
+@maxValue(168)
+@description('Optional. Retention cleanup policy. Number of hours to retain the tombstone markers of a compacted Event Hub. This value is only used when cleanupPolicy is Compact. Consumer must complete reading the tombstone marker within this specified amount of time if consumer begins from starting offset to ensure they get a valid snapshot for the specific key described by the tombstone marker within the compacted Event Hub.')
+param retentionDescriptionTombstoneRetentionTimeInHours int = 1
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
+param enableDefaultTelemetry bool = true
```

Comment:
Child modules should not have telemetry implemented.

Reply:
updated